src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,60 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { useContext } from 'react';
+import { TodosContext } from '../TodosContext/TodosContext';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC = () => {
+  const {
+    selectValue,
+    handleSelectValue,
+    handleChangeSearch,
+    searchValue,
+    clearSearchValue,
+  } = useContext(TodosContext);
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    handleSelectValue(event?.target.value.toLowerCase());
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectValue}
+            onChange={handleSelectChange}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={searchValue}
+          onChange={handleChangeSearch}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        {searchValue && (
```

Comment:
```suggestion
        {!!searchValue && (
```
in rendering with && for best rendering control use !! to return a boolean value